Replace combination of build-time smylinks and runtime moves with build time moves.

andreas-kupries opened the pull request:
Ref https://trello.com/c/H2AXtiHz/974-packages-mv-should-happen-at-build-time

Controlling SCF PR https://github.com/SUSE/scf/pull/2191

Drop symlinks for the role tarball.
Use `mv` in the Dockerfile instead.
Removed transform of symlinks into `mv` from the runtime role startup.

:white_check_mark: check unit tests
:white_check_mark: Get check build
:white_check_mark: Trial with SCF (and actual packages to move)

andreas-kupries committed:
[#H2AXtiHz] Replace combination of build-time smylinks and runtime moves with build time moves.

Drop symlinks for the role tarball.
Use `mv` in the Dockerfile instead.
Removed transform of symlinks into `mv` from the runtime role startup.

jandubois commented on builder/role_image.go:
```diff
@@ -300,14 +277,42 @@ func (r *RoleImageBuilder) generateDockerfile(instanceGroup *model.InstanceGroup
+	// because the symlinks will not resolve inside bpm containers.
+
+	packageSet := map[string]string{}
+	moves := []string{}
```
I don't like the name `moves`; these are really `commands` of any kind.

I can't make up my mind, but I think I might even prefer to have the `RUN` prefix in the template, and just add regular shell commands to a `runCommands` array that you expand with

```
{{ range .runCommands }}
RUN {{ . }}
{{ end }}
```

andreas-kupries replied:
Made this change, plus a few other fixes: Typo in the paths, and forgot to create destination directory.
Ref commit 6a73892c23d0eac50861ac6c031012d338d90a80

andreas-kupries committed:
Fixed typo in src/dst paths.
Fixed missing creation of dst.
Reworked to record only the shell commands, insert the `RUN` as part of the templating.
Renamed the variables to be more suitable.

Note: Exact form untested, some intermediates (fixes) locally tested and working.
